File: taskdesk-app/src/app/services/user.service.ts
import { TASK_DESK_API } from './taskdesk.api';
import { User } from '../model/User.model';
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable()
export class UserService {

  constructor(private http: HttpClient) { }

  login (user: User) {
    return this.http.post(`${TASK_DESK_API}/api/auth`, user);
  }

  createOrUpdate(user : User) {
    if (user.id != null && user.id != '' ){
      return this.http.put(`${TASK_DESK_API}/api/user`, user);
    } else {
      user.id = null;
      return this.http.post(`${TASK_DESK_API}/api/user`, user);
    }
  }
    
  
  findAll(page: number, count: number){
    return this.http.get(`${TASK_DESK_API}/api/user/${page}/${count}`);
  }

  findById(id: string) {
    return this.http.get(`${TASK_DESK_API}/api/user/${id}`);
  }

  delete(id: string) {
    return this.http.delete(`${TASK_DESK_API}/api/user/${id}`);
  }

    
  
}

File: TaskDesk/src/main/java/com/welbertsantos/taskdesk/api/service/UserService.java
package com.welbertsantos.taskdesk.api.service;

import org.springframework.data.domain.Page;
import org.springframework.stereotype.Component;

import com.welbertsantos.taskdesk.api.entity.User;

@Component
public interface UserService {
	
	User findByEmail(String email);
	
	User createOrUpdate(User usuario);
	
	User findById(String id);
	
	void delete(String id);
	
	Page<User> findAll(int page, int count);

}
